### ISSUE TITLE
Invalid Upstream = OK

### ISSUE THREAD

**Issue by MysticRyuujin** [emeraldpay/dshackle#176]
In the current release of Dshackle if I add an invalid upstream such as:

```
    - id: infura
      chain: ethereum
      role: secondary
      priority: 90
      options:
        disable-validation: true
      connection:
        ethereum:
          rpc:
            url: "https://mainnet.infura.io/v3/"
```

(notice it's missing an API key)

Dshackle just reports it as "OK" and shows no indication that there's anything wrong with the configured upstream.

**Comment by splix** [comment]
It's probably because of `disable-validation: true`. Can you try without it?

**Comment by MysticRyuujin** [comment]
It looks like, yes, setting `disable-validation` to `false` does result in it reporting the upstream as unavailable. However, that also means that the upstream has to support `net_peerCount` and not all of them do if you had a minimum peer count configured, correct?

**Comment by splix** [comment]
I've just added some extra options to configure validation in #183 
Now you can set `validate-peers` and `validate-syncing` individually. Will be available in v0.13.1 next week. Hope that solves the issue
